### DISCUSSION TITLE
Use GrabAngle on v6.0 and above?

### DISCUSSION THREAD

**Discussion by MarMi4** [leapmotion/UnityModules#1302] [Q&A] [answered]
Hi,

i just updated from v5.0.0 to v6.1.0 and cant get my head around an alternative for GrabAngle.

I used to compare hand.GrabAngle with a float to check if the hand is grabbing right now.
How can i do this without using GrabAngle now?

Thanks in advance and kind regards
Mike

**Comment by MattGrayUL** [comment, marked as answer]
Hi @MarMi4!

Have you tried using GrabStrength? It provides a value between 0 and 1 where 1 is fully grabbed and 0 is not grabbed at all, it may serve as easier to determine when the hand is grabbing.... (I would recommend using something like `if(hand.GrabStrength > 0.8f)` for a grab so it has a little cushion)

If not, here is a _similar_ but not exactly the same method for calculating 'grab angle':

```
float CalculateGrabAngle(Hand hand)
{
    // Compute the sum of the angles between the fingertips and hands.
    // For every finger, the angle is the sumb of bend + pitch + bow.
    float angleSum = 0.0f;
    for (int finger_idx = 1; finger_idx < 5; finger_idx++)
    {

        angleSum += Mathf.Lerp(0, Mathf.PI, hand.GetFingerStrength(finger_idx));
    }
    // Average between all fingers
    return angleSum / 4.0f;
}
```

**Reply by MarMi4** [reply]
Hi @MattGrayUL,

thank you for your fast answer.
I think GrabStrength should be enough for my purpose.
